Services/Curl.php

```diff
@@ -188,4 +193,33 @@ private function getError() {
         if (!curl_errno($this->curl)) return array();
         return array('error_no' => curl_errno($this->curl), 'error' => curl_error($this->curl));
     }
+    
+    /**
+     * Parse Http Headers in array
+     * 
+     * @param string $header
+     * @return Array
+     */
+     public static function httpParseHeaders($header) {
+         if (function_exists('http_parse_headers')) return http_parse_headers($header);
+
+         $retVal = array();
```

Comment:
I have not tested the it (the following code is just a sketch), but it should better look like:

```
$fields = explode("\r\n", preg_replace('/\x0D\x0A[\x09\x20]+/', ' ', $header));

return array_reduce($fields, function($carry, $field) {
    $match = [];
    if (!preg_match('/([^:]+): (.+)/m', $field, $match)) return $carry;

    $match[1] = preg_replace_callback('/(?<=^|[\x09\x20\x2D])./',function($matches) {
        return strtoupper($matches[0]);
    }, strtolower(trim($match[1])));

    $carry[$match[1]] = isset($carry[$match[1]]) ? [$carry[$match[1]], $match[2]] : trim($match[2]);
    return $carry;
}, []);
```